src/test/launchServer.ts

```diff
@@ -98,19 +80,19 @@ export class WebServer {
 
     launchedProcess.stderr.pipe(newProcessLogPrefixer()).pipe(process.stderr);
     launchedProcess.stdout.on('data', () => {});
+  }
 
-    if (this.config.port)
-      return this.config.port;
-    launchedProcess.stdout.on('data', collectPort);
-    const detectedPort = await Promise.race([
-      this._processExitedPromise,
-      collectPortPromise,
-    ]);
-    return detectedPort;
+  private async _waitForProcess() {
+    if (this.config.waitForPort) {
+      await this._waitForAvailability(this.config.waitForPort);
+      const baseURL = `http://localhost:${this.config.waitForPort}`;
+      process.env.PLAYWRIGHT_TEST_BASE_URL = baseURL;
```

Comment:
Is this documented? Should it be?

Reply:
This is considered as internal for now, not documented by intention.